src/Nethermind/Nethermind.Evm/EvmState.cs

```diff
@@ -107,8 +107,25 @@ public void ReturnStacks(byte[] dataStack, int[] returnStack)
 
         public int ReturnStackHead = 0;
         private bool _canRestore = true;
-
         public EvmState(
+            long gasAvailable,
+            ExecutionEnvironment env,
+            ExecutionType executionType,
+            bool isTopLevel,
```

Comment:
I think this should be removed as parameter and hardcoded to true, not to use this for not top level states

Reply:
removed